### DISCUSSION TITLE
Assign OP to selective phase from EBSD file

### DISCUSSION THREAD

**Discussion by ashishdhole** [idaholab/moose#26100] [Q&A Modules: Phase field]
Hi, I am Ashish, and I am very new to MOOSE. I am trying to study abnormal grain growth in the presence of insoluble particles in the micro structure. If I generate a synthetic micro structure EBSD file with phase ID as 1 for matrix and 2 for the particles. How can I assign my OP selectively to these two phases. Say a set of OP to phase 1 and a set of OP to phase 2. Thank you

**Comment by GiudGiud** [comment]
@laagesen @amjokisaari

**Comment by laagesen** [comment]
Hi Ashish, welcome to the MOOSE framework! Have a look at this documentation site for information on how to do this:

https://mooseframework.inl.gov/modules/phase_field/ICs/EBSD.html

**Reply by ashishdhole** [reply]
In the example files provided in the above link. Case 2. gives 1 OP per phase. That means if I have 10 grain of phase 1 and 10 grains of phase 2, I will have 2 OP associated with each phase. in Case 3. I can assign a set of OP to phase 1 and one single OP to phase 2. I think the PolycrystalICs action takes only one user-objects. If there was a way in which we can include two PolycrystalICs  corresponding to each phase, it will solve the issue. I want to know if it is possible? Is there any example file that uses such method. Looking forward to hearing from you. Thank you

**Reply by laagesen** [reply]
As long as the individual particles in the particle phase are separated from each other, you won't need separate order parameters for each one. A single order parameter should suffice for the particle phase.

**Reply by ashishdhole** [reply]
Thank you @laagesen for your reply. And that is true, separate particles can be assigned one order parameter. But lets say we have two phases and some insoluble particles. in assigning the order parameter for grains of each phase (which most of the time are adjacent to each other unlike insoluble particles which are separate). In simple words, my question is can we use PolycrystalICs action 'twice' in a single code with different conditions? If not, is there another way to do it? Also if you can provide a literature connected to materials/DeformedGrainMaterial, that will be great. Thank you.

**Comment by laagesen** [comment]
I am not sure if it's possible to use 2 different PolycrystalICs for different types of grains, @permcody do you know if this is possible?

**Reply by permcody** [reply]
You aren't going to want two PolycrystalICs for the separate phases. The data needed for each grain should be available in a single pass. This should be a function of the existing EBSD reader. @dschwen may know if this is possible already. If it's not, it shouldn't be too difficult to add this capability.
